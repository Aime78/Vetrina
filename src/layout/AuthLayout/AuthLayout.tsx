import {
  ScrollView,
  View,
  Text,
  Image,
  KeyboardAvoidingView,
  Platform,
} from 'react-native';
import styles from './AuthStyles';

type AuthLayoutProps = {
  title: string;
  description: string;
  children: React.ReactNode;
};

const AuthLayout = ({title, description, children}: AuthLayoutProps) => {
  return (
    <KeyboardAvoidingView
      style={styles.container}
      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
      <ScrollView keyboardDismissMode="on-drag">
        <View style={styles.imageContainer}>
          <Image
            style={styles.image}
            alt="logo"
            source={require('assets/logo.png')}
          />
        </View>
        <View style={styles.textContainer}>
          <Text style={styles.title}> {title}</Text>
          <Text style={styles.description}>{description}</Text>
        </View>
        <View>{children}</View>
      </ScrollView>
    </KeyboardAvoidingView>
  );
};

export default AuthLayout;
