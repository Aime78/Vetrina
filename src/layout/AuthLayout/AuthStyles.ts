import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    // width: '100%',
    // justifyContent: 'center',
    // alignItems: 'center',
    paddingTop: 80,
    paddingLeft: 15,
    paddingRight: 15,
  },
  imageContainer: {
    display: 'flex',
    alignItems: 'center',
  },
  image: {
    width: 155.1,
    height: 22.88,
  },
  textContainer: {
    marginTop: 30,
  },
  title: {
    textAlign: 'center',
    color: '#103B66',
    fontSize: 28,
    fontWeight: '500',
  },
  description: {
    marginTop: 5,
    textAlign: 'center',
    fontWeight: '400',
    fontSize: 18,
    color: '#546679',
  },
});

export default styles;
