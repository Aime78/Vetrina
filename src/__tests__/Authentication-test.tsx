import {Text} from 'react-native';
import React from 'react';
import AuthLayout from 'layout/AuthLayout/AuthLayout';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders the title and description', () => {
  const tree = renderer
    .create(
      <AuthLayout
        title="Title"
        description="Description"
        children={<Text>Children</Text>}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the children', () => {
  const tree = renderer
    .create(
      <AuthLayout
        title="Title"
        description="Description"
        children={<Text>Children</Text>}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
