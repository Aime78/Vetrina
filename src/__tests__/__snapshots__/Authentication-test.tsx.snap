// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the children 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 80,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    keyboardDismissMode="on-drag"
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <Image
          alt="logo"
          source={
            {
              "testUri": "../../../src/assets/logo.png",
            }
          }
          style={
            {
              "height": 22.88,
              "width": 155.1,
            }
          }
        />
      </View>
      <View
        style={
          {
            "marginTop": 30,
          }
        }
      >
        <Text
          style={
            {
              "color": "#103B66",
              "fontSize": 28,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          Title
        </Text>
        <Text
          style={
            {
              "color": "#546679",
              "fontSize": 18,
              "fontWeight": "400",
              "marginTop": 5,
              "textAlign": "center",
            }
          }
        >
          Description
        </Text>
      </View>
      <View>
        <Text>
          Children
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders the title and description 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 80,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    keyboardDismissMode="on-drag"
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <Image
          alt="logo"
          source={
            {
              "testUri": "../../../src/assets/logo.png",
            }
          }
          style={
            {
              "height": 22.88,
              "width": 155.1,
            }
          }
        />
      </View>
      <View
        style={
          {
            "marginTop": 30,
          }
        }
      >
        <Text
          style={
            {
              "color": "#103B66",
              "fontSize": 28,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
           
          Title
        </Text>
        <Text
          style={
            {
              "color": "#546679",
              "fontSize": 18,
              "fontWeight": "400",
              "marginTop": 5,
              "textAlign": "center",
            }
          }
        >
          Description
        </Text>
      </View>
      <View>
        <Text>
          Children
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
